ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AvailableStorageTable.tsx

```diff
@@ -273,7 +273,15 @@ const normaliseRowData = (
         ),
       },
       {
-        content: <DoubleRow primary={formatSize(storageDevice.size)} />,
+        content: (
+          <DoubleRow
+            primary={formatSize(storageDevice.size)}
+            secondary={
+              "available_size" in storageDevice &&
```

Comment:
It could be a nice improvement at some point to have a couple of variables at the top of the function for `isDisk` and `isPartition` to use in these cases so that it's a bit more obvious why you might not have these attributes.

Reply:
Yep, I'll be working on refactoring the storage components and making improvements for the rest of this week and next. I'll add this to the list!